jablib/src/test/java/org/jabref/logic/crawler/StudyYamlV1MigratorTest.java

```diff
@@ -0,0 +1,255 @@
+        assertEquals("simple query", simpleQuery.getQuery());
+
+        // Complex query
+        StudyQuery complexQuery = queries.get(1);
+        Map<String, String> expectedCatalogSpecific = Map.of("IEEE", "ieee specific query", "ACM", "acm specific query");
+        assertEquals(expectedCatalogSpecific, complexQuery.getCatalogSpecific());
+    }
+
+    @Test
+    void migrateWithNullEnabledDatabase() throws IOException {
+        String v1Content = """
+                authors:
+                  - "Author"
+                title: "Null Enabled Test"
+                research-questions:
+                  - "Question"
+                queries:
+                  - "query"
+                databases:
+                  - name: "Database1"
+                  - name: "Database2"
+                    enabled: true
+                  - name: "Database3"
+                    enabled: false
+                """;
+
+        Path studyFile = tempDir.resolve("study.yml");
+        Files.writeString(studyFile, v1Content);
+
+        Study migratedStudy = migrator.migrate(studyFile);
+
+        List<StudyCatalog> catalogs = migratedStudy.getCatalogs();
+        assertEquals(3, catalogs.size());
+
+        // First database should default to enabled=true
+        assertTrue(catalogs.getFirst().isEnabled());
+        assertEquals("Database1", catalogs.getFirst().getName());
+
+        // Second database explicitly enabled
+        assertTrue(catalogs.get(1).isEnabled());
+        assertEquals("Database2", catalogs.get(1).getName());
+
+        // Third database explicitly disabled
+        assertFalse(catalogs.get(2).isEnabled());
+        assertEquals("Database3", catalogs.get(2).getName());
+    }
+
+    @Test
+    void createMetadataSuccessfully() throws IOException {
+        String v1Content = """
+                authors:
+                  - "Author"
+                title: "Metadata Test"
+                research-questions:
+                  - "What is the research question?"
+                queries:
+                  - "test query"
+                databases:
+                  - name: "PubMed"
+                    enabled: true
+                  - name: "Embase"
+                    enabled: false
+                """;
+
+        Path studyFile = tempDir.resolve("study.yml");
+        Files.writeString(studyFile, v1Content);
+
+        Study migratedStudy = migrator.migrate(studyFile);
+
+        StudyMetadata metadata = migratedStudy.getMetadata().orElseThrow(() ->
+                new AssertionError("Expected metadata to be present after migration"));
+
+        // Check migration notes
+        String notes = metadata.getNotes().orElse("");
+        assertTrue(notes.contains("Migrated from v1.0 format"));
+        assertTrue(notes.contains("1 search queries"));
+        assertTrue(notes.contains("1 active databases"));
+    }
+
+    @Test
+    void migrateEmptyStudySuccessfully() throws IOException {
```

Comment:
Wrong approach, there should be two YAML files compared. Not input. YAML content and output: data model.